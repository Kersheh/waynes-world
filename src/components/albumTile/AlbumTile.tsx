import React from 'react';

import Button from 'components/button/Button';
import styles from './AlbumTile.module.scss';

interface AlbumTileProps {
  title: string;
  artist: string;
  // id?: string;
  // src?: any; // TODO
}
const AlbumTile = ({ title, artist }: AlbumTileProps) => {
  return (
    <Button styleType="tile">
      <div className={styles.albumTile}>
        <div className={styles.art} />

        <div className={styles.tag}>
          <div className={styles.title}>{title}</div>
          <div className={styles.artist}>{artist}</div>
        </div>
      </div>
    </Button>
  );
};

export default AlbumTile;
