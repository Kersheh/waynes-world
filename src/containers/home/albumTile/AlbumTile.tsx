import React, { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { parseISO, format } from 'date-fns';

import { fetchAlbumArt } from 'utils/serviceUtil';
import Button from 'components/button/Button';
import Image from 'components/image/Image';
import { openAlbumInLibraryAction } from '../homeActions';
import styles from './AlbumTile.module.scss';

interface AlbumTileProps {
  album: string;
  artist: string;
  id: string;
  createdAt: string;
}
const AlbumTile = ({ album, artist, id, createdAt }: AlbumTileProps) => {
  const dispatch = useDispatch();

  const [artBase64, setArtBase64] = useState<string | null>(null);

  // fetch potential album art from backend
  useEffect(() => {
    fetchAlbumArt(id, setArtBase64);
  }, [id]);

  return (
    <Button
      onClick={() => dispatch(openAlbumInLibraryAction(id))}
      styleType="tile"
    >
      <div className={styles.albumTile}>
        {!artBase64 && <div className={styles.art} draggable="false" />}
        {artBase64 && (
          <Image
            className={styles.art}
            src={`data:image/jpeg;base64,${artBase64}`}
          />
        )}

        <div className={styles.tag}>
          <div className={styles.album}>{album}</div>
          <div className={styles.artist}>{artist}</div>
          <div className={styles.timestamp}>
            Added {format(parseISO(createdAt), 'MMM dd, yyyy')}
          </div>
        </div>
      </div>
    </Button>
  );
};

export default AlbumTile;
