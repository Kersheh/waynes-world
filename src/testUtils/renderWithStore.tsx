import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose, Reducer } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { render } from '@testing-library/react';

import rootReducer from 'store/reducers';
import rootSaga from 'store/sagas';

const renderWithStore = (component: React.ReactElement) => {
  const sagaMiddleware = createSagaMiddleware();
  const store = createStore(
    rootReducer as Reducer,
    compose(applyMiddleware(sagaMiddleware))
  );
  sagaMiddleware.run(rootSaga);

  return render(<Provider store={store}>{component}</Provider>);
};

export default renderWithStore;
