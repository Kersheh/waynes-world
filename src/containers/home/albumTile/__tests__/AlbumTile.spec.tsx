import React from 'react';

import renderWithStore from 'testUtils/renderWithStore';
import AlbumTile from '../AlbumTile';

describe('<AlbumTile/> component', () => {
  it('should render', () => {
    const { container } = renderWithStore(
      <AlbumTile album="Album Title" artist="Artist" createdAt="2020-12-01T-01:00+00" id="1234567890" />
    );

    expect(container).toMatchSnapshot();
  });
});
