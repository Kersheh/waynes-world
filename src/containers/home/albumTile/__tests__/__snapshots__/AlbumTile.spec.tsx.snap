// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlbumTile/> component should render 1`] = `
<div>
  <button
    class="button tile"
    type="button"
  >
    <div
      class="albumTile"
    >
      <div
        class="art"
        draggable="false"
      />
      <div
        class="tag"
      >
        <div
          class="album"
        >
          Album Title
        </div>
        <div
          class="artist"
        >
          Artist
        </div>
        <div
          class="timestamp"
        >
          Added 
          Nov 30, 2020
        </div>
      </div>
    </div>
  </button>
</div>
`;
